src/Validation/Validator.php

```diff
@@ -0,0 +1,229 @@
+        }
+    }
+
+    /**
+     * @param  array $shipmentOptions
+     *
+     * @return array
+     */
+    private function convertNullValuesInShipmentOptions(array $shipmentOptions): array
+    {
+        foreach ($shipmentOptions as $shipmentOption => $value) {
+            if ('insurance' === $shipmentOption) {
+                $shipmentOptions[$shipmentOption] = (int) $value;
+                continue;
+            }
+
+            if ('labelDescription' === $shipmentOption) {
+                $shipmentOptions[$shipmentOption] = (string) $value;
+                continue;
+            }
+
+            $shipmentOptions[$shipmentOption] = (bool) $value;
+        }
+
+        return $shipmentOptions;
+    }
+
+    /**
+     * @param  string $what
+     * @param  string $column
+     * @param         $value
+     *
+     * @return $this
+     */
+    private function mergeIntoSchema(string $what, string $column, $value): self
+    {
+        if (! isset($this->extras[$what])) {
+            return $this;
+        }
+
+        $extras = $this->extras[$what];
+        $index  = array_search($value, array_column($extras, $column), true) ?: 0;
+        $extras = $extras[$index];
+
+        if (isset($extras['schema'])) {
+            $this->schema = $this->mergeSchemasRecursively($this->schema, $extras['schema']);
+        }
+
+        // todo implement json-schema draft 2019-09 or higher to integrate requirements into the schema validation
+        if (isset($extras['requirements'])) {
+            $this->requirements = $this->mergeSchemasRecursively($this->requirements, $extras['requirements']);
+        }
+
+        $this->extras = $extras;
+
+        return $this;
+    }
+
+    /**
+     * @param  array $schema
+     * @param  array $schema2
+     *
+     * @return array
+     */
+    private function mergeSchemasRecursively(array $schema, array $schema2): array
+    {
+        foreach ($schema2 as $key => $value) {
+            if (! isset($schema[$key]) || ! is_array($schema[$key])) {
+                $schema[$key] = [];
+            }
+
+            if (is_array($value) && 'enum' !== $key) {
+                $value = $this->mergeSchemasRecursively($schema[$key], $value);
+            }
+
+            $schema[$key] = $value;
+        }
+
+        return $schema;
+    }
+
+    /**
+     * @return array
+     */
+    public function getValidationSchema(): array
+    {
+        return $this->schema;
+    }
+
+    /**
+     * @return bool
+     * @throws \MyParcelNL\Pdk\Base\Exception\InvalidCastException
+     */
+    public function validate(): bool
+    {
+        $orderArray = $this->order->toArray();
+
+        $orderArray['deliveryOptions']['shipmentOptions'] = $this->convertNullValuesInShipmentOptions(
+            $orderArray['deliveryOptions']['shipmentOptions']
+        );
+
+        $validator = new JsonSchemaValidator();
+        $validator->validate($orderArray, $this->schema);
+
+        return $validator->isValid();
```

Comment:
ik mis nu de mogelijkheid om te kijken wat er precies fout was, dat gaan we wel heel hard nodig hebben